russstation/code/game/structures/stone.dm

```diff
@@ -0,0 +1,306 @@
+		deconstruct(TRUE)
+		user.visible_message(span_notice("[user] tears apart \the [src]."),
+			span_notice("You successfully disassemble \the [src]."),
+			span_hear("You hear stones dropping to the ground."))
+	else
+		return ..()
+
+/obj/structure/table/stone
+	name = "stone table"
+	desc = "A table made of smoothed stone. Ready to support feasts and drunken fights."
+	icon = 'russstation/icons/obj/smooth_structures/stone_table.dmi'
+	icon_state = "table-0"
+	base_icon_state = "table"
+	pass_flags_self = PASSTABLE | LETPASSTHROW
+	frame = /obj/structure/table_frame/stone
+	framestack = /obj/item/stack/sheet/mineral/stone
+	buildstack = /obj/item/stack/sheet/mineral/stone
+	buildstackamount = 1
+	framestackamount = 2
+	deconstruction_ready = FALSE
+	custom_materials = list(/datum/material/stone = 2000)
+	max_integrity = 100
+	integrity_failure = 0.33
+	smoothing_flags = SMOOTH_BITMASK
+	smoothing_groups = list(SMOOTH_GROUP_TABLES)
+	canSmoothWith = list(SMOOTH_GROUP_TABLES)
+
+/obj/structure/table/stone/deconstruction_hints(mob/user)
+	return "\The [src] would require some kind of <b>Dwarven tool</b> to deconstruct."
+
+/obj/structure/table/stone/attackby(obj/item/I, mob/living/user, params)
+	var/list/modifiers = params2list(params)
+	if(LAZYACCESS(modifiers, RIGHT_CLICK))
+		if(I.tool_behaviour == TOOL_SCREWDRIVER || I.tool_behaviour == TOOL_WRENCH)
+			to_chat(user, span_notice("You stare in bewilderment as [I] does nothing to [src]."))
+		else if (I.tool_behaviour == TOOL_DWARF)
+			to_chat(user, span_notice("You start deconstructing [src]..."))
+			I.play_tool_sound(src)
+			if(I.use_tool(src, user, 2 SECONDS, volume=50))
+				deconstruct(TRUE)
+				user.visible_message(span_notice("[user] tears apart \the [src]."),
+					span_notice("You successfully disassemble \the [src]."),
+					span_hear("You hear stones dropping to the ground."))
+	else
+		return ..()
+
+/obj/structure/chair/stone
+	name = "stone throne"
+	desc = "A throne made of stone. Gives your feet a rest while hurting your back."
+	icon = 'russstation/icons/obj/stone_structures.dmi'
+	icon_state = "chair"
+	resistance_flags = FIRE_PROOF
+	max_integrity = 250
+	integrity_failure = 0.1
+	custom_materials = list(/datum/material/stone = 2000)
+	layer = OBJ_LAYER
+	buildstacktype = /obj/item/stack/sheet/mineral/stone
+	buildstackamount = 1
+	item_chair = null
+
+/obj/structure/chair/stone/wrench_act(mob/living/user, obj/item/I)
+	to_chat(user, span_notice("You try using \the [I] but \the [src] says 'no'."))
+	return FALSE
+
+/obj/structure/chair/attackby(obj/item/I, mob/user, params)
+	if(I.tool_behaviour == TOOL_DWARF)
+		to_chat(user, span_notice("You begin deconstructing \the [src]..."))
+		I.play_tool_sound(src)
+		if(I.use_tool(src, user, 1 SECONDS, volume=30))
+			deconstruct()
+			user.visible_message(span_notice("[user] tears apart \the [src]."),
+				span_notice("You successfully disassemble \the [src]."),
+				span_hear("You hear stones dropping to the ground."))
+
```

Comment:
- This is overriding the base `obj/structure/chair/attackby` proc
- This doesn't call ever parent, which may cause funky behavior given attackby is a signal proc and is kinda important.